pkgs/test_core/lib/src/runner/vm/test_compiler.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:pool/pool.dart';
+import 'package:test_api/backend.dart'; // ignore: deprecated_member_use
+import 'package:frontend_server_client/frontend_server_client.dart';
+
+import '../package_version.dart';
+import '../../util/package_config.dart';
+
+class CompilationResponse {
+  final String? compilerOutput;
+  final int errorCount;
+  final Uri? kernelOutputUri;
+
+  CompilationResponse(
+      {this.compilerOutput, this.errorCount = 0, this.kernelOutputUri});
+}
+
+class TestCompiler {
+  final _compilePool = Pool(1);
+  final String _dillCachePath;
+  final Directory _outputDillDirectory;
+
+  late final _outputDill =
+      File(p.join(_outputDillDirectory.path, 'output.dill'));
+  FrontendServerClient? _frontendServerClient;
+
+  TestCompiler(this._dillCachePath)
```

Comment:
[nit] Can we add a comment either here or on the class level about how instantiating this object doesn't start the process and it instead waits until the first request? That makes it more clear in the usage location that we don't need to worry about waiting.

Reply:
done